### ISSUE TITLE
Division by zero exception when getSourceImageDimensions() fails

### ISSUE THREAD

**Issue by mijewe** [spacecatninja/craft-imager-x#52]
### I'm submitting a... 

- [x] bug report
- [ ] feature request
- [ ] general question

### Steps to reproduce

1. Attempt to transform an image from an external URL, which fails to return a sourceWidth and sourceHeight.
1a. An example is https://epc.rightmove.co.uk/generateEpc?graphtype=EPC&currentenergy=51&potentialenergy=51&currentenvironment=60&potentialenvironment=60&graphquality=web
2. See a "Division by zero" exception

### Description
If `getSourceImageDimensions()` fails, it returns `(null,null)` (or perhaps `(0,0)`, I'm not sure).
This causes `if ($sourceWidth / $sourceHeight > $paramsW / $paramsH) {` to throw a "Division By Zero" Exception.

I'm not sure what the behaviour should be here, but throwing an exception at this point a bit harsh for a production site.

### Displayed errors, stack trace, relevant logs  
```
yii\base\ErrorException: Division by zero
#39 /xxx/vendor/spacecatninja/imager-x/src/models/ImgixTransformedImageModel.php(86): handleError
#38 /xxx/vendor/craftcms/cms/src/web/ErrorHandler.php(76): handleError
#37 /xxx/vendor/spacecatninja/imager-x/src/models/ImgixTransformedImageModel.php(86): __construct
#36 /xxx/vendor/spacecatninja/imager-x/src/transformers/ImgixTransformer.php(116): getTransformedImage
#35 /xxx/vendor/spacecatninja/imager-x/src/transformers/ImgixTransformer.php(67): transform
#34 /xxx/vendor/spacecatninja/imager-x/src/services/ImagerService.php(421): transformImage
#33 /xxx/vendor/spacecatninja/imager-x/src/variables/ImagerVariable.php(37): transformImage
#32 /xxx/vendor/twig/twig/src/Extension/CoreExtension.php(1499): twig_get_attribute
#31 /xxx/vendor/craftcms/cms/src/helpers/Template.php(90): attribute
#30 /xxx/storage/runtime/compiled_templates/ae/aed0d388e6eda568848d2a362521a05eba6db42339feff059e75cb3026b9609b.php(137): doDisplay
#29 /xxx/vendor/twig/twig/src/Template.php(407): displayWithErrorHandling
#28 /xxx/vendor/twig/twig/src/Template.php(380): display
#27 /xxx/storage/runtime/compiled_templates/0e/0e7cffe2332472b476592d49c76cba72075603ce8700fa43a41b8bd1540c33cd.php(68): doDisplay
#26 /xxx/vendor/twig/twig/src/Template.php(407): displayWithErrorHandling
#25 /xxx/vendor/twig/twig/src/Template.php(380): display
#24 /xxx/storage/runtime/compiled_templates/fb/fbe7bc3c5061c6af93d9bbf44203beb78ca38f6ad7dea79e01481507f90e586d.php(165): block_main
#23 /xxx/vendor/twig/twig/src/Template.php(184): displayBlock
#22 /xxx/storage/runtime/compiled_templates/7b/7b5f351e866bdc1377d5ae805240ee87c56f49eb4eed00c742a5bbb34e40bbcd.php(130): doDisplay
#21 /xxx/vendor/twig/twig/src/Template.php(407): displayWithErrorHandling
#20 /xxx/vendor/twig/twig/src/Template.php(380): display
#19 /xxx/storage/runtime/compiled_templates/fb/fbe7bc3c5061c6af93d9bbf44203beb78ca38f6ad7dea79e01481507f90e586d.php(137): doDisplay
#18 /xxx/vendor/twig/twig/src/Template.php(407): displayWithErrorHandling
#17 /xxx/vendor/twig/twig/src/Template.php(380): display
#16 /xxx/vendor/twig/twig/src/Template.php(392): render
#15 /xxx/vendor/twig/twig/src/TemplateWrapper.php(45): render
#14 /xxx/vendor/twig/twig/src/Environment.php(318): render
#13 /xxx/vendor/craftcms/cms/src/web/View.php(391): renderTemplate
#12 /xxx/vendor/craftcms/cms/src/web/View.php(452): renderPageTemplate
#11 /xxx/vendor/craftcms/cms/src/web/Controller.php(243): renderTemplate
#10 /xxx/vendor/craftcms/cms/src/controllers/TemplatesController.php(97): actionRender
#9 /xxx/vendor/craftcms/cms/src/controllers/TemplatesController.php(0): call_user_func_array
#8 /xxx/vendor/yiisoft/yii2/base/InlineAction.php(57): runWithParams
#7 /xxx/vendor/yiisoft/yii2/base/Controller.php(180): runAction
#6 /xxx/vendor/craftcms/cms/src/web/Controller.php(181): runAction
#5 /xxx/vendor/yiisoft/yii2/base/Module.php(528): runAction
#4 /xxx/vendor/craftcms/cms/src/web/Application.php(294): runAction
#3 /xxx/vendor/yiisoft/yii2/web/Application.php(103): handleRequest
#2 /xxx/vendor/craftcms/cms/src/web/Application.php(279): handleRequest
#1 /xxx/vendor/yiisoft/yii2/base/Application.php(386): run
#0 index.php(21): null
```

### Additional info

- Imager version & edition: Pro v3.1.3
- Imager transformer: imgix
- Craft version: Craft Pro 3.5.5
- PHP version: 7.3.11
- Image driver & version: GD 7.2.31
- Database driver & version: MySQL 5.5.5
- Plugins & versions (if relevant):

**Comment by aelvan** [comment]
Thanks for reporting. This was fixed in 3.2.2.

**Comment by mijewe** [comment]
Hi - I'm having this same issue again.

`getSourceImageDimensions` is returning `(null,null)`, which is passing the check `if ($sourceWidth !== 0 && $sourceHeight !== 0) {` (since null !== 0), so I'm hitting the same exception at `if ($sourceWidth / $sourceHeight > $paramsW / $paramsH) {`

Cheers
